packages/@sanity/form-builder/src/inputs/SelectInput.tsx

```diff
@@ -26,15 +26,34 @@ const SelectInput = React.forwardRef(function SelectInput(
   const validation = markers.filter((marker) => marker.type === 'validation')
   const errors = validation.filter((marker) => marker.level === 'error')
 
+  const itemFromOptionValue = (optionValue) => {
+    const index = Number(optionValue)
+
+    return items[index]
+  }
+
+  const optionValueFromItem = (item) => {
+    return items.indexOf(item)
+  }
+
   const inputId = useId()
   const handleChange = React.useCallback(
-    (nextItem) => {
+    (nextItem: TitledListValue<string | number>) => {
       onChange(
         PatchEvent.from(typeof nextItem.value === 'undefined' ? unset() : set(nextItem.value))
       )
     },
     [onChange]
   )
+
+  const handleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const nextItem = itemFromOptionValue(event.currentTarget.value)
+
+    if (nextItem) {
+      handleChange(nextItem)
+    }
```

Comment:
We need an `else` condition to set the value back to `undefined`, something like:

```js
} else {
  handleChange(EMPTY_ITEM)
}
```

Reply:
Good point! Added in 32f22ed19